### PR TITLE
VB-2760 Prisoner released events are processed

### PR DESCRIPTION
**Pull request by AledWEvans**
Prisoner released events are processed

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitNotificationEventRepository.kt [hunk 1]
```diff
@@ -12,7 +12,7 @@ interface VisitNotificationEventRepository : JpaRepository<VisitNotificationEven
   @Query(
     "SELECT count(ve) > 0" +
       " FROM visit_notification_event ve " +
-      " WHERE ve.create_timestamp BETWEEN NOW() - INTERVAL '1 MINUTE' AND NOW() " +
+      " WHERE ve.create_timestamp BETWEEN NOW() - INTERVAL '10 MINUTE' AND NOW() " +
       " AND ve.booking_reference=:bookingReference AND ve.type=:#{#notificationEvent.name()}",
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitNotificationEventRepository.kt:16]: updated this as we were getting duplicates in preprod

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt [hunk 2]
```diff
@@ -187,6 +187,21 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
     @Param("endDateTime") endDateTime: LocalDateTime? = null,
   ): Boolean
 
+  @Query(
+    "SELECT v  FROM Visit v LEFT JOIN v.visitors as vis " +
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:191]: join with visitors can be removed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt [hunk 3]
```diff
@@ -494,4 +494,8 @@ class VisitService(
 
     return visitCancellationDateAllowed
   }
+
+  fun getFutureVisitsBy(prisonerNumber: String, prisonCode: String): List<VisitDto> {
+    return this.visitRepository.getVisits(prisonerNumber, prisonCode, LocalDateTime.now()).map { VisitDto(it) }
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt:499]: It's for Future visit so need to specify

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**